app/scripts/controllers/edit/buildConfig.js

```diff
@@ -117,6 +118,75 @@ angular.module('openshiftConsole')
       "SerialLatestOnly"
     ];
 
+    $scope.buildHookTypes = [{
+      id: "command",
+      label: "Command"
+    }, {
+      id: "script",
+      label: "Shell Script"
+    }, {
+      id: "args",
+      label: "Arguments to default image entry point"
+    }, {
+      id: "commandArgs",
+      label: "Command with arguments"
+    }, {
+      id: "scriptArgs",
+      label: "Shell script with arguments"
+    }];
+
+    $scope.buildHookSelection = {
+      type: {}
+    };
+
+    var getInitialBuildHookSelection = function() {
+      var hasArgs = !_.isEmpty(_.get($scope, 'buildConfig.spec.postCommit.args'));
+      var hasCommand = !_.isEmpty(_.get($scope, 'buildConfig.spec.postCommit.command'));
+      var hasScript = !!_.get($scope, 'buildConfig.spec.postCommit.script');
+      $scope.view.hasHooks = hasArgs || hasCommand || hasScript;
+
+      if (hasArgs && hasCommand) {
+        $scope.buildHookSelection.type = _.find($scope.buildHookTypes, { id: 'commandArgs' });
+      } else if (hasArgs && hasScript) {
+        $scope.buildHookSelection.type = _.find($scope.buildHookTypes, { id: 'scriptArgs' });
+      } else if (hasArgs) {
+        $scope.buildHookSelection.type = _.find($scope.buildHookTypes, { id: 'args' });
+      } else if (hasScript) {
+        $scope.buildHookSelection.type = _.find($scope.buildHookTypes, { id: 'script' });
+      } else {
+        $scope.buildHookSelection.type = _.find($scope.buildHookTypes, { id: 'command' });
+      }
```

Comment:
Nit: I'd assign the id to a var and the only do the _.find in one place. Then avoids repeating the same code.

```js
var id;
if (hasArgs && hasCommand) {
  id = 'commandArgs';
} else if ...

$scope.buildHookSelection.type = _.find($scope.buildHookTypes, { id: id });

```